### DISCUSSION TITLE
Visual Studio Code Intellisense doesn't work on macOS

### DISCUSSION THREAD

**Discussion by AndaBucsa** [cake-build/cake#3171] [Q&A] [answered]
Hello,

I am seeing the following error in OmniSharp log:

```
[fail]: OmniSharp.Cake.CakeProjectSystem
        /Users/cristina-andabucsa/Desktop/test-cake/build.cake will be ignored due to an following error
System.IO.FileNotFoundException: Cake is not installed. Path /Users/cristina-andabucsa/Desktop/test-cake/tools/Cake/Cake.Core.dll does not exist.
  at OmniSharp.Cake.CakeProjectSystem.GetProject (Cake.Scripting.Abstractions.Models.CakeScript cakeScript, System.String filePath) [0x00051] in <4cc10c29fc0b4bfca54acb8eb39811ca>:0 
  at OmniSharp.Cake.CakeProjectSystem.AddCakeFile (System.String cakeFilePath) [0x0001f] in <4cc10c29fc0b4bfca54acb8eb39811ca>:0 
```

I am not sure why it searches for Cake in the current directory, as I have installed Cake via `dotnet tool install --global Cake.Tool`.

This is my current setup:

```
macOS 10.15.6 (19G2021)

Visual Studio Code
Version: 1.53.2

dotnet cake --version
1.0.0

dotnet --version
5.0.102

cake-build.cake-vscode
1.0.0

ms-dotnettools.csharp
1.23.9
```

I also have Visual Studio for Mac / Mono / Xamarin installed.

I have created an empty directory and ran the following commands from Visual Studio Code:

```
Install to workspace (without a bootstrap file, cake.config) and with .NET Tool (Global) debugging dependencies
Install intellisense support
```

The same steps on a different macOS Big Sur machine works, so I'm pretty sure it's something wrong with my specific setup, but unfortunately I'm not sure what to try any more.

Thank you,
Anda

**Comment by augustoproiete** [comment, marked as answer]
I have a similar issue on my Mac and I suspect it's related to the VS Code extension for Cake assuming that the `dotnet-cake` tool should be available in the path (and it seems it is when I run on the terminal), but something on my setup is causing `dotnet-cake` not to be available in the path when the VS Code extension for Cake tries to run it, at least. It needs more investigation.

In the meantime, @bjorkstromm provided me with a workaround that I've been able to use for all my projects, and you might want to try it out:

From VS Code:

1. (`command` + `shift` + `p` on Mac) or (`ctrl` + `shift` + `p` on Windows)
2. Cake: Install debug dependencies
3. Cake: Install intellisense support (select the Cake runner for .NET Core)
4. Restart OmniSharp (or restart VS Code)

![image](https://user-images.githubusercontent.com/177608/108428861-9e42f500-7215-11eb-8932-b026d688697f.png)

![image](https://user-images.githubusercontent.com/177608/108428878-a1d67c00-7215-11eb-85da-18ab41fb94db.png)

![image](https://user-images.githubusercontent.com/177608/108428892-a6029980-7215-11eb-9ec3-e0b4db6d933f.png)

**Reply by nils-a** [reply]
@augustoproiete didn't know you had the problem in VSCode, too. Should we try to get someting similar to https://github.com/cake-build/cake-rider/issues/78 implemented in VSCode?

**Reply by augustoproiete** [reply]
@nils-a Yes, I think it makes sense to give that a try.

**Reply by nils-a** [reply]
It seems it was really late yesterday.. Upon reading this again, the problem seemingly is not in starting the runner from VSCode but in O# or rather bakery finding and accessing it. I have opened https://github.com/cake-build/bakery/issues/125 to track this.

**Reply by AndaBucsa** [reply]
The workaround fixed the issue, thank you.

I noticed however that Find All References / CodeLense doesn't report usages from other files, it only reports usages from the  same file.

**Reply by augustoproiete** [reply]
@AndaBucsa Great that the workaround fixes the issue. We're tracking the fix to use the .NET tool in https://github.com/cake-build/cake-vscode/issues/509 (which depends on https://github.com/cake-build/bakery/issues/125 as @nils-a mentioned above).

The CodeLens issue you mentioned might not related, and needs more investigation. Tracking via https://github.com/cake-build/cake-vscode/issues/510
